src/vsphere_cpi/lib/cloud/vsphere/attach_tag_to_vm.rb

```diff
@@ -0,0 +1,134 @@
+require 'cloud/vsphere/logger'
+require 'vsphere-automation-cis'
+require 'vsphere-automation-vcenter'
+
+module  VSphereCloud
+  module TaggingTag
+    class AttachTagToVm
+      include Logger
+
+      def initialize(api_client)
+        @api_client = api_client
+      end
+
+      private def tag_assoc_info
```

Comment:
Why are these privately cached objects for the class? Can you just create them when needed?

Reply:
You would want to cache something if it is repeatedly accessed.

Reply:
Done.